unlock-app/src/__tests__/reducers/walletStatusReducer.test.js

```diff
@@ -0,0 +1,18 @@
+import reducer from '../../reducers/walletStatusReducer'
+import { WAIT_FOR_WALLET, GOT_WALLET } from '../../actions/walletStatus'
+
+describe('walletStatusReducer', () => {
+  const notWaiting = { waiting: false }
+  const waiting = { waiting: true }
+  it('should return the initial state', () => {
+    expect(reducer(undefined, {})).toEqual(notWaiting)
+  })
+
+  it('should set waiting to true when receiving WAIT_FOR_WALLET', () => {
+    expect(reducer(notWaiting, { type: WAIT_FOR_WALLET })).toEqual(waiting)
+  })
+
+  it('should set waiting to false when receiving GOT_WALLET', () => {
+    expect(reducer(waiting, { type: GOT_WALLET })).toEqual(notWaiting)
```

Comment:
```suggestion
    expect(reducer(waiting, { type: GOT_WALLET })).toBe(initialState)
```

All the changes I'm suggesting are to check we are not mutating state when we don't need to.